scrapy/loader/__init__.py

```diff
@@ -127,13 +129,12 @@ def get_value(self, value, *processors, **kw):
         return value
 
     def load_item(self):
-        item = self.item
+        adapter = ItemAdapter(self.item)
         for field_name in tuple(self._values):
             value = self.get_output_value(field_name)
             if value is not None:
-                item[field_name] = value
-
-        return item
+                adapter[field_name] = value
+        return adapter.item
```

Comment:
```suggestion
        return self.item
```

Reply:
It's the same object, and I'd argue that `return adapter.item` is a little bit more clear in the sense that in order to understand `return self.item` you need to know that `ItemAdapter` modifies the item in-place, which you might not know a priori.

Reply:
I think that it’s precisely because it’s important for people to understand that `ItemAdapter` edits the item in place that we should avoid `adapter.item` unless necessary.

That said, this is internal code and not documentation, so I guess that aspect is not that relevant.